File: dropwizard-scala_2.9.1/src/test/scala/com/yammer/dropwizard/scala/params/tests/LongParamTest.scala
package com.yammer.dropwizard.scala.params.tests

import org.junit.Test
import javax.ws.rs.WebApplicationException
import com.simple.simplespec.Spec
import com.yammer.dropwizard.scala.params.LongParam

class LongParamTest extends Spec {

  class `A valid long parameter` {
    private val param = LongParam("40")

    @Test def `has an int value` = {
      param.value.must(be(40L))
    }
  }

  class `An invalid long parameter` {
    @Test def `throws a WebApplicationException with an error message` = {
      evaluating {
        LongParam("poop")
      }.must(throwAnExceptionLike {
        case e: WebApplicationException => {
          val response = e.getResponse
          response.getStatus.must(be(400))
          response.getEntity.must(be("Invalid parameter: poop (Must be an integer value.)"))
        }
      })
    }
  }

}

File: dropwizard-core/src/test/java/com/yammer/dropwizard/jersey/params/tests/LongParamTest.java
package com.yammer.dropwizard.jersey.params.tests;

import com.yammer.dropwizard.jersey.params.LongParam;
import org.junit.Test;

import javax.ws.rs.WebApplicationException;
import javax.ws.rs.core.Response;

import static org.fest.assertions.api.Assertions.assertThat;
import static org.fest.assertions.api.Assertions.failBecauseExceptionWasNotThrown;

public class LongParamTest {
    @Test
    public void aLongReturnsALong() throws Exception {
        final LongParam param = new LongParam("200");

        assertThat(param.get())
                .isEqualTo(200L);
    }

    @Test
    @SuppressWarnings("ResultOfObjectAllocationIgnored")
    public void aNonIntegerThrowsAnException() throws Exception {
        try {
            new LongParam("foo");
            failBecauseExceptionWasNotThrown(WebApplicationException.class);
        } catch (WebApplicationException e) {
            final Response response = e.getResponse();

            assertThat(response.getStatus())
                    .isEqualTo(400);

            assertThat((String) response.getEntity())
                    .isEqualTo("\"foo\" is not a number.");
        }
    }
}
